api/src/main/java/openconsensus/metrics/DoubleGauge.java

```diff
@@ -75,33 +76,12 @@
  * @since 0.1.0
  */
 @ThreadSafe
-public interface DoubleGauge extends Metric {
+public interface DoubleGauge extends Metric<TimeSeries> {
```

Comment:
Talking to @bogdandrutu, the reason we have separate classes for double and long gauges/cumulatives instead of a single generic class is to avoid `Int` and `Double` types and the headache of java type boxing. Is there a better way around this?